drivers/rtc/rtc_ll_stm32.c

```diff
@@ -146,6 +146,7 @@ static int rtc_stm32_init(const struct device *dev)
 
 	/* Enable RTC clock source */
 	if (clock_control_configure(clk, (clock_control_subsys_t)&cfg->pclken[1], NULL) != 0) {
+		z_stm32_hsem_unlock(CFG_HW_RCC_SEMID);
```

Comment:
Are you proposing to remove the `hsem` completely? Otherwise, I would keep the missing unlock in the code. As I wrote above I'm not sure if I understand your proposal correctly.

Reply:
The lock/unlock around `LL_PWR_EnableBkUpAccess` isn't required

Reply:
Ok, I removed them, sorry for that misunderstanding.